lib/dialects/abstract/query-generator.js

```diff
@@ -2240,6 +2240,11 @@ var QueryGenerator = {
 
       if (escapeValue) {
         value = this.escape(value, field, escapeOptions);
+
+        //if ANY is used with like, add parentheses to generate correct query
+        if (escapeOptions.acceptStrings && (comparator.indexOf('ANY') > comparator.indexOf('LIKE'))) {
+         value = '(' + value + ')';
+        }
```

Comment:
I can add such logic [here](https://github.com/sushantdhiman/sequelize/blob/fix-5736/lib/dialects/abstract/query-generator.js#L2220), but it will require checking if `key` is either one of `$like, $iLike, $notLike, $notiLike` , which again will be better done using `indexOf` check. Because checking for all of them by if `==` conditions will be too verbose.

While I can move logic to proposed position, it will need same check (`indexOf` match). I think code will be more simple in current way, away from `comparator` creation process. 

Please suggest what you think @mickhansen